cudf/series.py

```diff
@@ -181,9 +181,14 @@ def __len__(self):
 
     def __getitem__(self, arg):
         if isinstance(arg, Series):
-            selvals, selinds = columnops.column_select_by_boolmask(
+            if arg.dtype == np.int64:
+                selvals, selinds = columnops.column_select_by_position(
                 self._column, arg)
-            index = self.index.take(selinds.to_gpu_array())
+                index = self.index.take(selinds.to_gpu_array())
+            else:
```

Comment:
I think it would be better to check if the dtype is boolean and do it this way, otherwise do it by position.